src/test/java/com/puppycrawl/tools/checkstyle/AuditEventDefaultFormatterTest.java

```diff
@@ -80,6 +83,40 @@ public void testCalculateBufferLength() throws Exception {
         assertEquals(54, result, "Buffer length is not expected");
     }
 
+    @Test
+    public void testFormatModuleNameContainsCheckSuffix() {
+        final AuditEvent mock = mock(AuditEvent.class);
+        when(mock.getSourceName()).thenReturn("TestModuleCheck");
+        when(mock.getSeverityLevel()).thenReturn(SeverityLevel.WARNING);
```

Comment:
why we do mock if we can just create AuditEvent with required values ?

Reply:
To cover [this](https://github.com/checkstyle/checkstyle/blob/master/src/main/java/com/puppycrawl/tools/checkstyle/AuditEventDefaultFormatter.java#L106) line we need a class that has no package and ends with `Check` suffix.

This can be done by adding a class to the test source root.

Or, by extracting this code to some utility method.
Such method can be reused [here](https://github.com/checkstyle/checkstyle/blob/master/src/main/java/com/puppycrawl/tools/checkstyle/checks/SuppressWarningsHolder.java#L161), [here](https://github.com/checkstyle/checkstyle/blob/master/src/main/java/com/puppycrawl/tools/checkstyle/checks/coding/IllegalTypeCheck.java#L737), [here](https://github.com/checkstyle/checkstyle/blob/master/src/main/java/com/puppycrawl/tools/checkstyle/checks/coding/IllegalTypeCheck.java#L752) and possible in many other methods.

Reply:
>This can be done by adding a class to the test source root.

Let's do this.
Without such comment, it is very very unclear.
Previously we agreed that if we do some reflection based test we explain in comment what is a goal. 
I think mocking tests should be commented also. Ideally no mocking tests.

Reply:
I see comment over methods, but I would be awesome to avoid mocking if that is simple as you describe.

I am approving PR.

Reply:
Please review https://github.com/checkstyle/checkstyle/pull/11056